src/io/parsers/SDFParser.js

```diff
@@ -0,0 +1,304 @@
+    }
+  }
+
+  _parseMOL(stream) {
+    this._compoundIndx++;
+
+    this._parseHeader(stream);
+    const countsLine = stream.getStringFromStart(3);
+    const atomsNum = parseInt(countsLine.substr(0, 3), 10);
+    const bondsNum = parseInt(countsLine.substr(3, 3), 10);
+    this._parseAtoms(stream, atomsNum);
+    this._parseBonds(stream, bondsNum);
+
+    this._atomsParsed += atomsNum;
+
+    this._metadata.molecules[this._compoundIndx]._residues = [];
+    this._metadata.molecules[this._compoundIndx]._residues.push(this._residue);
+  }
+
+  _parseDataItem(stream) {
+    const tag = stream.getCurrentString();
+
+    let data = [];
+    let curStr = stream.getNextString();
+
+    //read data
+    while (curStr.trim() !== '') {
+      data.push(curStr);
+      curStr = stream.getNextString();
+    }
+    if (data.length === 1) {
+      data = data[0];
+    }
+    const property = {tag: tag.replace(/<|>/g, '').trim(), data: data};
+    this._currentMolProps.push(property);
+  }
+
+  _parseCompound(stream) {
+    this._parseMOL(stream);
+
+    //parse data items block
+    if (this._format === fileFormat.SDF) {
+      while (stream.findNextDataItem()) {
+        this._parseDataItem(stream);
+      }
+      if (this._currentMolProps.length !== 0) {
+        const molecule = this._metadata.molecules[this._compoundIndx];
+        molecule.props = this._currentMolProps;
+        this._tryToUpdateMoleculeData(molecule);
+      }
+    }
+  }
+
+  _fixBondsArray() {
+    const serialAtomMap = this._serialAtomMap;
+    const complex = this._complex;
+
+    const bonds = complex._bonds;
+    for (let j = 0; j < bonds.length; j++) {
+      const bond = bonds[j];
+      if (bond._right < bond._left) {
+        console.log('_fixBondsArray: Logic error.');
+      }
+      bond._left = serialAtomMap[bond._left] || null;
+      bond._right = serialAtomMap[bond._right] || null;
+    }
+  }
+
+  _buildAssemblies() {
+    const chains = this._complex._chains;
+
+    if (chains.length === 1) {
+      return this._assemblies;
+    }
+
+    for (let i = 0; i < chains.length; i++) {
+      const assembly = new Assembly(this._complex);
+      let matrix = new THREE.Matrix4();
+      assembly.addMatrix(matrix);
+      assembly.addChain(chains[i]._name);
+      this._assemblies.push(assembly);
+    }
+
+    return this._assemblies;
+  }
+
+  _buildMolecules() {
+    this._complex._molecules = [];
+    const molecules = this._metadata.molecules;
+    for (let i = 0; i < molecules.length; i++) {
+      const molecule = new Molecule(this._complex, molecules[i].name, i + 1);
+      molecule._residues = molecules[i]._residues;
+      this._complex._molecules[i] = molecule;
+    }
+
+    return this._complex._molecules;
+  }
+
+  _searchTag(tag, props) {
+    for (let i = 0; i < props.length; i++) {
+      if ((tag instanceof RegExp && tag.test(props[i].tag)) || tag === props[i].tag) {
+        return props[i].data;
+      }
+    }
+    return false;
```

Comment:
Let's keep the return value type consistent. Is it really a boolean? What signature do you expect this method to have? Shouldn't it be `null` object or `undefined` value instead?